Source/WebKit/Platform/IPC/HandleMessage.h

```diff
@@ -321,8 +331,13 @@ void handleMessage(C& connection, Decoder& decoder, T* object, MF U::* function)
     static_assert(std::is_same_v<typename ValidationType::MessageArguments, typename MessageType::Arguments>);
 
     auto arguments = decoder.decode<typename MessageType::Arguments>();
-    if (!arguments) [[unlikely]]
+    if (!arguments) [[unlikely]] {
+#if ENABLE(IPC_TESTING_API)
+        if constexpr (std::is_same_v<C, IPC::Connection>)
+            setConnectionErrorStringIfNeeded(connection, decoder);
```

Comment:
This should not be here, not all messages are processed with handle message.
You can see Connection.cpp, search for m_ignoreInvalidMessageForTesting. There are the locations roughly where you need to convert message from decoder into message from connection.